From d8d9a42d7e243f0874f77c0fc932f32ecbde42e6 Mon Sep 17 00:00:00 2001
From: Eduardo Habkost <ehabkost@redhat.com>
Date: Fri, 26 Apr 2019 17:41:04 -0300
Subject: [PATCH] STX: Add support statement to -help output

RH-Author: Eduardo Habkost <ehabkost@redhat.com>
Message-id: <1386183197-27761-1-git-send-email-ehabkost@redhat.com>
Patchwork-id: 55994
O-Subject: [qemu-kvm RHEL7 PATCH] Add support statement to -help output
Bugzilla: 972773
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Acked-by: knoel@redhat.com
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>

Bugzilla: 972773
Brew scratch build:
http://brewweb.devel.redhat.com/brew/taskinfo?taskID=6676272

Add support statement to -help output, reporting direct qemu-kvm usage
as unsupported by Red Hat, and advising users to use libvirt instead.

Change the above to say StarlingX instead of Red Hat.
STX-Author: Jim Somerville <Jim.Somerville@windriver.com>

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
Signed-off-by: Rafael Falcao <Rafael.VieiraFalcao@windriver.com>
---
 softmmu/vl.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/softmmu/vl.c b/softmmu/vl.c
index e6e0ad5a92..27ce96f671 100644
--- a/softmmu/vl.c
+++ b/softmmu/vl.c
@@ -1688,9 +1688,17 @@ static void version(void)
            QEMU_COPYRIGHT "\n");
 }
 
+static void print_stx_warning(void)
+{
+    printf("\nWARNING: Direct use of qemu-kvm from the command line is not supported by StarlingX.\n"
+             "WARNING: Use libvirt as the stable management interface.\n"
+             "WARNING: Some command line options listed here may not be available in future releases.\n\n");
+}
+
 static void help(int exitcode)
 {
     version();
+    print_stx_warning();
     printf("usage: %s [options] [disk_image]\n\n"
            "'disk_image' is a raw hard disk image for IDE hard disk 0\n\n",
             error_get_progname());
@@ -1707,6 +1715,7 @@ static void help(int exitcode)
            "\n"
            QEMU_HELP_BOTTOM "\n");
 
+    print_stx_warning();
     exit(exitcode);
 }
 
-- 
2.25.1
